qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,521 @@
+       If `expand_number_op` is `True`, then factors of `N` are expanded.
+
+    Raises:
+       ValueError: if any character in `term_str` is not one of "+-IN".
+    """
+    (n_number, n_raise, n_lower) = (0, 0, 0)
+    facs = []
+    for i, c in enumerate(term_str):
+        if c == "I":
+            continue
+        if c == "+":
+            n_raise += 1
+            facs.append((i, "+"))
+        elif c == "-":
+            n_lower += 1
+            facs.append((i, "-"))
+        elif c == "N":
+            n_number += 1
+            if expand_number_op:
+                facs.append((i, "+"))
+                facs.append((i, "-"))
+            else:
+                facs.append((i, "N"))
+        else:
+            raise ValueError("Unexpected operator ", c, " in term.")
+
+    return (n_number, n_raise, n_lower), facs
+
+
+def _interaction_type(n_number: int, n_raise: int, n_lower: int) -> str:
+    """
+    Return a string describing the type of interaction given the number of
+    number, raising, and lowering operators.
+
+    The types of interaction returned are 'number', 'excitation', 'coulomb_exchange',
+    'number_excitation', 'double_excitation'.
+
+    Args:
+       `n_number`: the number of number operators
+       `n_raise`: the number of raising operators
+       `n_lower`: the number of lowering operators
+
+    Returns:
+      str: One of 'number', 'excitation', 'coulomb_exchange',
+      'number_excitation', 'double_excitation'.
+
+    Raises:
+      ValueError: if the numbers of operators don't describe a one- or two-body term from
+      an electronic Hamiltonian.
+    """
+    if n_raise == 0 and n_lower == 0:
+        if n_number == 1:
+            return "number"
+        elif n_number == 2:
+            return "coulomb_exchange"
+        else:
+            raise ValueError("unexpected number of number operators")
+    elif n_raise == 1 and n_lower == 1:
+        if n_number == 1:
+            return "number_excitation"
+        elif n_number == 0:
+            return "excitation"
+        else:
+            raise ValueError("unexpected number of number operators")
+    elif n_raise == 2 and n_lower == 2:
+        return "double_excitation"
+    else:
+        raise ValueError("unexpected number of operators")
+
+
+def number_of_modes(fer_op: FermionicOp) -> int:
+    """Return the number of modes (including identities) in each term `fer_op`"""
+    return len(fer_op.to_list()[0][0])
+
+
+def operator_string(term: tuple) -> str:
+    """
+    Return the string describing the operators in the term extracted from a `FermionicOp`.
+    given by `term.
+    """
+    return term[0]
+
+
+def operator_coefficient(term: tuple) -> float:
+    """
+    Return the coefficient of the multi-mode operator term extracted from a `FermionicOp`.
+    """
+    return term[1]
+
+
+## TODO: We may want a lower-triangular matrix. This may be the cause of the minus sign error.
```

Comment:
What's the status on this?

Reply:
This is a stale comment. I have now removed it